dotnet/src/Functions/Functions.Prompty/Extensions/PromptyKernelExtensions.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.PromptTemplates.Handlebars;
+using Microsoft.SemanticKernel.PromptTemplates.Liquid;
+using Microsoft.SemanticKernel.Prompty.Core;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.SemanticKernel.Prompty.Extension;
+
+/// <summary>
+/// Extension methods for <see cref="Kernel"/> to create a <see cref="KernelFunction"/> from a prompty file.
+/// </summary>
+public static class PromptyKernelExtensions
+{
+    /// <summary>
+    /// Create a <see cref="KernelFunction"/> from a prompty file.
+    /// </summary>
+    /// <param name="kernel">kernel</param>
+    /// <param name="promptyPath">path to prompty file.</param>
+    /// <param name="promptTemplateFactory">prompty template factory, if not provided, a <see cref="LiquidPromptTemplateFactory"/> will be used.</param>
+    /// <param name="loggerFactory">logger factory</param>
+    /// <returns><see cref="KernelFunction"/></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    /// <exception cref="NotSupportedException"></exception>
+    public static KernelFunction CreateFunctionFromPrompty(
+        this Kernel kernel,
+        string promptyPath,
+        IPromptTemplateFactory? promptTemplateFactory = null,
+        ILoggerFactory? loggerFactory = null)
+    {
+        Verify.NotNull(kernel);
+
+        var text = File.ReadAllText(promptyPath);
+
+        promptTemplateFactory ??= new AggregatorPromptTemplateFactory(new HandlebarsPromptTemplateFactory(), new LiquidPromptTemplateFactory());
+
+        // create PromptTemplateConfig from text
+        // step 1
+        // retrieve the header, which is in yaml format and put between ---
+        //
+        // e.g
+        // file: chat.prompty
+        // ---
+        // name: Contoso Chat Prompt
+        // description: A retail assistant for Contoso Outdoors products retailer.
+        // authors:
+        //   - XXXX
+        // model:
+        //   api: chat
+        //   configuration:
+        //     type: azure_openai
+        //     azure_deployment: gpt - 35 - turbo
+        //     api_version: 2023 - 07 - 01 - preview
+        //   parameters:
+        //     tools_choice: auto
+        //     tools:
+        //     -type: function
+        //       function:
+        //         name: test
+        //         description: test function
+        //         parameters:
+        //           properties:
+        //             location:
+        //               description: The city and state or city and country, e.g.San Francisco, CA
+        //                 or Tokyo, Japan
```

Comment:
Before a function can be invoked by SK, it must be imported into SK. During the import, SK will automatically extract all possible metadata, including name, description, parameters and return type schemas.  
   
It would be helpful to understand scenarios when the SK metadata is insufficient and when the prompt metadata should be used instead of the SK one, or the other way around.

CC: @matthewbolanos

Reply:
Thanks, that's why the tool support is not considered in this PR because it's not fully compatible with how sk modeling function call. We need further discussion on that part.

Reply:
Track in #6034